sycl/source/detail/pi_opencl.cpp

```diff
@@ -59,9 +59,35 @@ pi_result OCL(piDevicesGet)(pi_platform      platform,
   return cast<pi_result>(result);
 }
 
+// Entry type, matches OpenMP for compatibility, used in binary selection
+struct __tgt_offload_entry {
+  void *addr;
+  char *name;
+  size_t size;
+  int32_t flags;
+  int32_t reserved;
+};
+
```

Comment:
Since we started using the symbol tables in SYCL, the proper way to fix this would be redefining the _pi_offload_entry type in the pi.h from ```typedef void * _pi_offload_entry``` to

```
struct _pi_offload_entry_struct {
  void *addr;
  char *name;
  size_t size;
  int32_t flags;
  int32_t reserved;
};
typedef _pi_offload_entry_struct * _pi_offload_entry;
```
and using ```_pi_offload_entry``` in your code instead of ```__tgt_offload_entry*```